src/modules/media/media.controller.ts

```diff
@@ -37,6 +40,24 @@ export class MediaController {
         return this.mediaService.getById(id)
     }
 
+    @Post('search')
+    @ApiParam({ name: 'companyId', type: String })
+    @ApiBody({ type: MediaSearchBodyDTO })
+    @ApiOkResponse({ type: MediaResponsePaginatedDTO })
+    @ApiQuery({ name: 'query', type: String, required: false })
+    search(
+        @Param('companyId') companyId: string,
+        @Body() body: MediaSearchBodyDTO,
+        @Query() query?: string,
```

Comment:
```suggestion
        @Query('query') query?: string,
```
Esse você não deve ter testado